include/galsim/SBInterpolatedImageImpl.h

```diff
@@ -175,6 +194,56 @@ namespace galsim {
         void operator=(const SBInterpolatedImageImpl& rhs);
     };
 
+
+    class SBInterpolatedKImage::SBInterpolatedKImageImpl :
+        public SBInterpolated::SBInterpolatedImpl
+    {
+    public:
+
+        template <typename T>
+        SBInterpolatedKImageImpl(
+            const BaseImage<T>& realKImage,
+            const BaseImage<T>& imagKImage,
+            double dk, double stepk,
+            boost::shared_ptr<Interpolant2d> kInterp,
+            const GSParamsPtr& gsparams);
+
+        ~SBInterpolatedKImageImpl();
+
+        std::complex<double> kValue(const Position<double>& p) const;
+        // void fillKValue(tmv::MatrixView<std::complex<double> > val,
+        //                 double kx0, double dkx, int izero,
+        //                 double ky0, double dky, int jzero) const;
+        // void fillKValue(tmv::MatrixView<std::complex<double> > val,
+        //                 double kx0, double dkx, double dkxy,
+        //                 double ky0, double dky, double dkyx) const;
+
+        ConstImageView<double> getRealKImage() const;
+        ConstImageView<double> getImagKImage() const;
+
+        double xValue(const Position<double>& p) const
+        { throw SBError("SBInterpolatedKImage::xValue() is not implemented"); }
+
+        Position<double> centroid() const;
+
+        boost::shared_ptr<PhotonArray> shoot(int N, UniformDeviate u) const
+        { throw SBError("SBInterpolatedKImage::shoot() is not implemented"); }
+
+    protected:  // Made protected so that these can be used in the derived CorrelationFunction class
+
+        std::string repr() const;
+        double _dk;
+        int Ninitial;
+        int Nk;
+        mutable double xcentroid;
+        mutable double ycentroid;
```

Comment:
These should have leading underscores.  Our style in the C++ layer has been that member variables get a leading underscore to mark them as such.  Kind of like the `self.` prefix in the python layer for member variables.  But shorter.